structure_tests/structure_test_v1.go

```diff
@@ -127,18 +137,48 @@ func ProcessCommand(t *testing.T, fullCommand []string, checkOutput bool) (strin
 	if stderr != "" {
 		t.Logf("stderr: %s", stderr)
 	}
+	var exitCode int
 	if err != nil {
 		if checkOutput {
 			// The test might be designed to run a command that exits with an error.
 			t.Logf("Error running command: %s. Continuing.", err)
 		} else {
 			t.Fatalf("Error running setup/teardown command: %s.", err)
 		}
+		exitCode = err.(*exec.ExitError).Sys().(syscall.WaitStatus).ExitStatus()
+	} else {
+		exitCode = cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()
+	}
+	ResetEnvVars(t, originalVars)
```

Comment:
would that give the test function enough time to complete with the vars set though? I'm not totally sure how defer works, but I thought it would run immediately after the first function finished (so in this case it would set and then immediately unset the vars)